src/js/share.ts

```diff
@@ -0,0 +1,66 @@
+/* global document, navigator, window */
+import { CityId } from "./types";
+import { determineShareUrl } from "./cityId";
+/**
+ * Copy `value` to the user's clipboard
+ *
+ * @param string value
+ */
+const copyToClipboard = async (value: string) => {
+  try {
+    await navigator.clipboard.writeText(value);
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error("Failed to write to clipboard: ", err);
+  }
+};
+
+/**
+ * Toggle share link icon briefly to show user an indicator
+ *
+ * @param {HTMLAnchorElement} shareIcon
+ */
+const switchIcons = (shareIcon: HTMLAnchorElement) => {
+  const linkIcon: HTMLAnchorElement | null = shareIcon.querySelector(
+    "svg.share-link-icon"
+  );
+  const checkIcon: HTMLAnchorElement | null = shareIcon.querySelector(
+    "svg.share-check-icon"
+  );
+  if (linkIcon && checkIcon) {
+    linkIcon.style.display = "none";
+    checkIcon.style.display = "inline-block";
+    setTimeout(() => {
+      linkIcon.style.display = "inline-block";
+      checkIcon.style.display = "none";
+    }, 1000);
+  }
+};
+
+/**
+ * Add an event listener for the share button to copy the link to the clipboard.
+ *
+ * @param string cityId: e.g. `st.-louis-mo`
+ */
+const setUpShareUrlClickListener = (cityId: CityId) => {
+  // We put the event listener on `map` because it is never erased, unlike the copy button
+  // being recreated every time the score card changes. This is called "event delegation".
+  const mapElement: HTMLElement | null = document.querySelector("#map");
+  if (mapElement) {
```

Comment:
You can invert the conditional. If `!mapElement`, then `return`. It avoids this indentation.

Reply:
Same as the other comment.  I like that this makes it explicit that mapElement within the braces is no longer ```HTMLElement | null``` but rather is ```HTMLElement``` after narrowing the type via the conditional.